enhancements/sig-policy/122-expanded-hub-templates-access/README.md

```diff
@@ -0,0 +1,310 @@
+
+```go
+func (r *ReplicatedPolicyReconciler) getToken(ctx context.Context, namespace string, saName string) (string, error) {
+	var expiration int64 = 7200
+
+	req := &authv1.TokenRequest{
+		Spec: authv1.TokenRequestSpec{
+			// TODO: Is this always the correct value?
+			Audiences: []string{"https://kubernetes.default.svc.cluster.local"},
+			// 2 hours
+			ExpirationSeconds: &expiration,
+		},
+	}
+
+	returnedTokenReq, err := r.HubClient.CoreV1().ServiceAccounts(namespace).CreateToken(
+		ctx, saName, req, metav1.CreateOptions{},
+	)
+	if err != nil {
+		return "", err
+	}
+
+	tokenFile, err := os.CreateTemp("", fmt.Sprintf("token-%s.%s-*", namespace, saName))
+	if err != nil {
+		return "", err
+	}
+
+	_, err = tokenFile.Write([]byte(returnedTokenReq.Status.Token))
+	if err != nil {
+		return "", err
+	}
+
+	tokenFilePath, err := filepath.Abs(tokenFile.Name())
+	if err != nil {
+		return "", err
+	}
+
+	expirationTimestamp := returnedTokenReq.Status.ExpirationTimestamp
+
+	go func() {
+		defer func() {
+			err := os.Remove(tokenFilePath)
+			if err != nil {
+				log.Error(err, "Failed to clean up the service account token file", "serviceAccount", saName, "namespace", namespace, "path", tokenFilePath)
+			}
+		}()
+
+		for {
+			deadlineCtx, deadlineCtxCancel := context.WithDeadline(ctx, expirationTimestamp.Add(-10*time.Minute))
+
+			<-deadlineCtx.Done()
+			if !errors.Is(deadlineCtx.Err(), context.DeadlineExceeded) {
+				// This really does nothing but this satisfies the linter that cancel is called.
+				deadlineCtxCancel()
+
+				return
+			}
+
+			returnedTokenReq, err := r.HubClient.CoreV1().ServiceAccounts(namespace).CreateToken(
+				ctx, saName, req, metav1.CreateOptions{},
+			)
+			if err != nil {
+				log.Error(err, "Failed to renew the token. Will retry in 5 seconds.", "serviceAccount", saName, "namespace", namespace)
+
+				time.Sleep(5 * time.Second)
+
+				continue
+			}
+
+			_, err = tokenFile.Write([]byte(returnedTokenReq.Status.Token))
+			if err != nil {
+				log.Error(err, "Failed to write the token. Will retry in 5 seconds.", "serviceAccount", saName, "namespace", namespace, "path", tokenFilePath)
+
+				time.Sleep(5 * time.Second)
+
+				continue
+			}
+
+			expirationTimestamp = returnedTokenReq.Status.ExpirationTimestamp
+
+			// This really does nothing but this satisfies the linter that cancel is called.
+			deadlineCtxCancel()
+		}
+	}()
+
+	return tokenFilePath, nil
+}
+```
+
+### Risks and Mitigation
+
+### Open Questions [optional]
+
+None yet.
+
+### Test Plan
+
+**Note:** _Section not required until targeted at a release._
+
+### Graduation Criteria
+
+N/A
+
+### Upgrade / Downgrade Strategy
+
+Downgrades would cause the `spec.serviceAccountName` field to be removed and fall back to the default service account.
+
+### Version Skew Strategy
+
+## Implementation History
+
+N/A
+
+## Drawbacks
+
+- Complex implementation
```

Comment:
Requires management of an additional service account by the policy author.  This makes the policy development a little more complex, but trades the benefit of not copying resources to the policy authoring namespace.